cocos/2d/CCNode.h

```diff
@@ -1881,9 +1895,17 @@ class CC_DLL Node : public Ref
     mutable bool _additionalTransformDirty; ///< transform dirty ?
     bool _transformUpdated;         ///< Whether or not the Transform object was updated since the last frame
 
-    int _localZOrder;               ///< Local order (relative to its siblings) used to sort the node
+    union {
+        struct {
+            std::int32_t z; // The original localZOrder
+            std::uint32_t a; // Order Of Arrival, for avoid sort problem with unstable_sort algorithm.
+        } detail;
+        std::int64_t value; // The value to be used in sort
+    } _localZOrder;               ///< Local order (relative to its siblings) used to sort the node
```

Comment:
@halx99 i have merged the reverted PR. So you can send a new PR now, thanks. And because we start to test v3.13, and the PR will modify the basic codes of cocos2d-x, so i may not include your PR in v3.13 but next version.